stdlib/array.gr

```diff
@@ -1014,9 +1014,55 @@ provide let rotate = (n, arr) => {
   }
 }
 
+let rec _chunk = (chunkSize, allowPartial, arr, arrLen, chunkStartIndex) => {
+  if (arrLen == 0 || arrLen < chunkSize && !allowPartial) {
+    []
+  } else if (arrLen < chunkSize) {
+    [slice(chunkStartIndex, length(arr), arr)]
+  } else {
+    // create the first chunk of the given array
+    let firstChunk = slice(chunkStartIndex, chunkStartIndex + chunkSize, arr)
+    let restChunks = _chunk(
+      chunkSize,
+      allowPartial,
+      arr,
+      arrLen - chunkSize,
+      chunkStartIndex + chunkSize
+    )
+    [firstChunk, ...restChunks]
+  }
+}
+
+/**
+ * Splits the given array into chunks of the provided size.
+ *
+ * @param chunkSize: The size of each chunk
+ * @param allowPartial: Whether to allow partial chunks at the end of the array
+ * @param arr: The array to chunk
+ * @returns An array of chunks
+ *
+ * @throws InvalidArgument(String): When `chunkSize` is not an integer
+ * @throws InvalidArgument(String): When `chunkSize` is less than zero
+ *
+ * @example chunk(2, false, [> 1, 2, 3, 4]) == [> [> 1, 2], [> 3, 4] ]
+ * @example chunk(2, false, [> 1, 2, 3, 4, 5]) == [> [> 1, 2], [> 3, 4] ]
+ * @example chunk(2, true, [> 1, 2, 3, 4, 5]) == [> [> 1, 2], [> 3, 4], [> 5] ]
+ * @since v0.6.0
+ */
+provide let chunk = (chunkSize, allowPartial, arr) => {
```

Comment:
I don't like this `allowPartial` argument

Reply:
Really? I could for see cases where `allowPartial` could come in hand.

Reply:
It's not that it's not useful, but for a general purpose function in the standard library, it can get confusing seeing the behavior change via this boolean. If we wanted to support both behaviors, then it should be two separate functions. That said though, I think we can just have the partial behavior (doesn't lose data and is predictable).

Reply:
Fair enough I will make that change.